File: BaseBall Game Leetcode 682/Solution.java
import java.util.Stack;

public class Solution{
    public static void main(String[] args) {
        Solution1 sol = new Solution1();
        String[] ops = {"5","2","C","D","+"};
        int score = sol.calPoints(ops);
        System.out.println(score);

    }
}
class Solution1 {
    public int calPoints(String[] operations) {
        Stack<Integer> stack = new Stack<>();

        for (String op : operations) {
            if (op.equals("+")) {
                int top = stack.pop();
                int newTop = top + stack.peek();
                stack.push(top);
                stack.push(newTop);
            } else if (op.equals("D")) {
                stack.push(stack.peek() * 2);
            } else if (op.equals("C")) {
                stack.pop();
            } else {
                stack.push(Integer.parseInt(op));
            }
        }

        int sum = 0;
        while (!stack.isEmpty()) {
            sum += stack.pop();
        }
        
        return sum;
    }
}

File: BaseBall Game Leetcode 682/Solution.py
class Solution:
    def calPoints(self, operations):
        stack = []
        for op in operations:
            if op == '+':
                stack.append(stack[-1]+stack[-2])
            elif op == 'D':
                stack.append(stack[-1]*2)
            elif op == 'C':
                stack.pop()
            else:
                stack.append(int(op))
        return sum(stack)
    
sol = Solution()
ops = ["5","2","C","D","+"]
print(sol.calPoints(ops))
    